packages/ra-core/src/controller/input/ReferenceArrayInputController.spec.tsx

```diff
@@ -417,6 +417,44 @@ describe('<ReferenceArrayInputController />', () => {
         assert.equal(crudGetMany.mock.calls.length, 2);
     });
 
+    it('should call crudGetMany when input value changes only with the additional input values', () => {
+        const crudGetMany = jest.fn();
+        const wrapper = shallow(
+            <ReferenceArrayInputController
+                {...defaultProps}
+                input={{ value: [5] }}
+                allowEmpty
+                crudGetMany={crudGetMany}
+            />
+        );
+        expect(
+            crudGetMany.mock.calls[crudGetMany.mock.calls.length - 1]
+        ).toEqual([defaultProps.reference, [5]]);
+        wrapper.setProps({ input: { value: [5, 6] } });
+        expect(
+            crudGetMany.mock.calls[crudGetMany.mock.calls.length - 1]
+        ).toEqual([defaultProps.reference, [6]]);
+    });
+
+    it('should call crudGetMany with empty list when already fetched input value changes', () => {
```

Comment:
No, it should not call crudGetMany at all in that case.

Reply:
Ah ok, that's what I did at first but it broke previous tests. I will edit this